File: Algorithms/Easy/1299_ReplaceElementsWithGreatestElementOnRightSide/Solution.cs
public class Solution {
    public int[] ReplaceElements(int[] arr) {
        int size = arr.Length, maxVal = -1;
        int[] answer = new int[size];
        
        for (int i = size - 1; i >= 0; i--) {
            answer[i] = maxVal;
            maxVal = Math.Max(maxVal, arr[i]);
        }
        
        return answer; 
    }
}

File: Algorithms/Easy/1299_ReplaceElementsWithGreatestElementOnRightSide/Solution.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int size = arr.size(), maxVal = -1;
        vector<int> answer (size, 0);
        
        for (int i = size - 1; i >= 0; i--) {
            answer[i] = maxVal;
            maxVal = max(maxVal, arr[i]);
        }
        
        return answer;
    }
};